openhcl/underhill_attestation/src/crypto.rs

```diff
@@ -130,8 +134,26 @@ pub fn pkcs11_rsa_aes_key_unwrap(
     wrapped_key_blob: &[u8],
 ) -> Result<Rsa<Private>, Pkcs11RsaAesKeyUnwrapError> {
     let modulus_size = unwrapping_rsa_key.size();
-    let wrapped_aes_key = &wrapped_key_blob[..modulus_size as usize];
-    let wrapped_rsa_key = &wrapped_key_blob[modulus_size as usize..];
+
+    let mut blob_chunks = wrapped_key_blob.chunks_exact(modulus_size as usize);
+    let wrapped_aes_key = blob_chunks.next().ok_or_else(|| {
+        Pkcs11RsaAesKeyUnwrapError::UndersizedWrappedKeyBlob(format!(
+            "expected wrapped AES key blob to be {} bytes, but found {}",
+            modulus_size,
+            blob_chunks.remainder().len()
+        ))
+    })?;
+    let wrapped_rsa_key = blob_chunks.next().ok_or_else(|| {
```

Comment:
Is this correct to assume two wrapped key blobs have the same size? Can we have test with actual payload?

Reply:
Ah, I see you add the test in the other PR. Have you test the two changes together?

Reply:
Not until now and I see that this change isn't quite right - the key blob is not `2 * modulus_size` after all. I'll adjust my change

Reply:
Updated in https://github.com/microsoft/openvmm/pull/587/commits/a4e505248aaa4f7eb871059138d4847340f9522e